src/components/dialog/dialog.scss

```diff
@@ -0,0 +1,358 @@
+
+  /**
+    * @prop --ch-dialog-max-block-size:
+    * Specifies the maximum block size of the dialog. Useful for scenarios where the
+    * dialog is resizable.
+    * @default auto
+    */
+  --ch-dialog-max-block-size: auto;
+
+  /**
+      * @prop --ch-dialog-max-inline-size:
+      * Specifies the maximum inline size of the dialog. Useful for scenarios
+      * where the dialog is resizable.
+      * @default auto
+      */
+  --ch-dialog-max-inline-size: auto;
+
+  /**
+    * @prop --ch-dialog-min-block-size:
+    * Specifies the minimum block size of the dialog. Useful for scenarios where the
+    * dialog is resizable.
+    * @default auto
+    */
+  --ch-dialog-min-block-size: auto;
+
+  /**
+     * @prop --ch-dialog-min-inline-size:
+     * Specifies the minimum inline size of the dialog. Useful for scenarios
+     * where the dialog is resizable.
+     * @default auto
+     */
+  --ch-dialog-min-inline-size: auto;
+
+  /**
+    * @prop --ch-dialog-resize-threshold:
+    * Specifies the size of the threshold to resize the dialog.
+    * @default 4px
+    */
+  --ch-dialog-resize-threshold: 8px;
+
+  /**
+   * @prop --ch-dialog-separation-x:
+   * Specifies the separation between the action and dialog in the x axis.
+   * @default 0px
+   */
+  --ch-dialog-separation-x: 0px;
+
+  /**
+   * @prop --ch-dialog-separation-y:
+   * Specifies the separation between the action and dialog in the y axis.
+   * @default 0px
+   */
+  --ch-dialog-separation-y: 0px;
+
+  --ch-dialog-resize-threshold--half-negative: calc(
+    var(--ch-dialog-resize-threshold) * -0.5
+  );
+
+  --ch-dialog-border-inline-start-width: 0px;
+  --ch-dialog-border-inline-end-width: 0px;
+  --ch-dialog-border-block-start-width: 0px;
+  --ch-dialog-border-block-end-width: 0px;
+
+  --ch-dialog-resize__inline-start: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-inline-start-width) * 0.5
+  );
+
+  --ch-dialog-resize__inline-end: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-inline-end-width) * 0.5
+  );
+
+  --ch-dialog-resize__block-start: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-block-start-width) * 0.5
+  );
+
+  --ch-dialog-resize__block-end: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-block-end-width) * 0.5
+  );
+
+  --ch-dialog-dragged-x: 0px;
+  --ch-dialog-dragged-y: 0px;
+  --ch-dialog-rtl: 1;
+
+  // Necessary when the dialog is not placed in a new top layer. For example,
+  // when nesting dropdowns
+  position: fixed;
+
+  // Reset browser defaults
+  margin: 0;
+  padding: 0;
+  background-color: unset;
+  border: unset;
+  color: unset;
+  overflow: visible;
+
+  top: 0;
+  inset-inline-start: 0;
+}
+
+// - - - - - - - - - - - - - - - -
+//           Dialog
+// - - - - - - - - - - - - - - - -
+dialog {
+  // Reset browser default
+  margin: unset;
+
+  position: fixed;
```

Comment:
- position "fixed" is a reset, since the dialog position value is "absolute" when is not a modal.
- overflow "visible" is a reset, since the dialog overflow value is "auto" when the dialog is a modal.